### PR TITLE
[WebApp] Add protected action page

### PR DESCRIPTION
**Pull request by Maks19**
**Description:** Add protected action page

**Issue:** https://www.notion.so/saladtech/Protected-Action-page-3a516fccc3474ceea00c92847e950ce9

image

### CODE REVIEW

File: packages/web-app/src/modules/protected-action/ProtectedActionPageContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { connect } from '../../connect'
+import type { RootStore } from '../../Store'
+import { ProtectedActionPage } from './components'
+
+const mapStoreToProps = (store: RootStore): any => ({
+  isPasskeySupported: store.passkey.isPasskeySupported,
+  backToPreviousPage: () => store.routing.goBack(),
+  verifyWithBackupCode: () => {},
+  verifyWithPasskey: () => {},
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/ProtectedActionPageContainer.tsx:9]: https://github.com/SaladTechnologies/salad-applications/pull/1168#discussion_r1711245350

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+const _ProtectedActionPage: FC<Props> = ({
+  classes,
+  isPasskeySupported,
+  backToPreviousPageClick,
+  verifyWithBackupCodeClick,
+  verifyWithPasskeyClick,
+}) => {
+  const handleVerifyWithBackupCodeSubmit = (values: FormValues) => {
+    verifyWithBackupCodeClick?.(values.backupCode)
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:103]:
```suggestion
    verifyBackupCode?.(values.backupCode)
```

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+                    </Field>
+                    <div>
+                      <Button variant="primary-basic" label="Verify with Backup Code" type="submit" />
+                    </div>
+                  </form>
+                </div>
+              )
+            }}
+          />
+          <Button variant="outlined" label="Cancel" onClick={backToPreviousPageClick} />
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:182]:
```suggestion
          <Button variant="outlined" label="Cancel" onClick={onBackToPreviousPageClick} />
```

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            <Text className={classes.description} variant="baseL">
+              This device does not support Passkeys. Please login on a device or browser that supports passkeys and try
+              again.
+            </Text>
+          )}
+          <Text className={classes.description} variant="baseL">
+            Or verify with a one-time backup code:
+          </Text>
+          <Form
+            onSubmit={handleVerifyWithBackupCodeSubmit}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:154]:
```suggestion
            onSubmit={handleVerifyBackupCodeSubmit}
```

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+          {isPasskeySupported ? (
+            <>
+              <Text className={classes.description} variant="baseL">
+                Please verify your credentials to proceed with this action.
+              </Text>
+              <Button
+                leadingIcon={<FontAwesomeIcon icon={faKey} className={classes.addPasskeyIcon} />}
+                variant="primary-basic"
+                label="Verify with Passkey"
+                onClick={verifyWithPasskeyClick}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:141]:
```suggestion
                onClick={onVerifyPasskeyClick}
```

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  verifyWithBackupCodeClick: (backupCode: string) => void
+  verifyWithPasskeyClick: () => void
+}
+
+const _ProtectedActionPage: FC<Props> = ({
+  classes,
+  isPasskeySupported,
+  backToPreviousPageClick,
+  verifyWithBackupCodeClick,
+  verifyWithPasskeyClick,
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:100]:
```suggestion
  onVerifyPasskeyClick,
```

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+  isPasskeySupported: boolean
+  backToPreviousPageClick: () => void
+  verifyWithBackupCodeClick: (backupCode: string) => void
+  verifyWithPasskeyClick: () => void
+}
+
+const _ProtectedActionPage: FC<Props> = ({
+  classes,
+  isPasskeySupported,
+  backToPreviousPageClick,
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:98]:
```suggestion
  onBackToPreviousPageClick,
```

**Review by vitto-moz — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx (hunk 2 above)
Comment (Maks19) [reply packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:103]: Any reason to change it?

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx (hunk 4 above)
Comment (Maks19) [reply packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:154]: Why do we need to remove `With`?

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx (hunk 2 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:103]: the functionality is about verifying the code itself, right?

File: packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx (hunk 4 above)
Comment (vitto-moz) [reply packages/web-app/src/modules/protected-action/components/ProtectedActionPage.tsx:154]: the functionality is about verifying the code itself, right?

File: packages/web-app/src/modules/protected-action/ProtectedActionPageContainer.tsx [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { connect } from '../../connect'
+import type { RootStore } from '../../Store'
+import { ProtectedActionPage } from './components'
+
+const mapStoreToProps = (store: RootStore): any => ({
+  isPasskeySupported: store.passkey.isPasskeySupported,
+  onBackToPreviousPageClick: () => store.routing.goBack(),
+  verifyWithBackupCodeClick: () => {},
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/ProtectedActionPageContainer.tsx:8]:
```suggestion
  verifyWithBackupCode: () => {},
```

File: packages/web-app/src/modules/protected-action/ProtectedActionPageContainer.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+  onVerifyPasskeyClick: () => {},
```
Comment (vitto-moz) [inline packages/web-app/src/modules/protected-action/ProtectedActionPageContainer.tsx:9]:
```suggestion
  onVerifyWithPasskeyClick: () => {},
```

**Review by vitto-moz — CHANGES_REQUESTED** [review]

**Review by vitto-moz — APPROVED** [review]

**Pull request merged**